index.js

```diff
@@ -200,173 +215,329 @@ while (args.length) {
+		await page.setViewport({
+			width: width,
+			height: height || 800
 		});
-	}
 
-	debug.loadTime = new Date() - debug.loadTime;
+		if (stripResources) {
+			await page.setRequestInterception(true);
+
+			var baseUrl = url || fakeUrl;
+		
+			page.on("request", request => {
+		
+				var _url = request.url();
+		
+				if (_url.indexOf(baseUrl) > -1) {
+					_url = _url.slice(baseUrl.length);
+				}
+		
+				if (outputDebug && !_url.match(/^data/) && debug.requests.indexOf(_url) < 0) {
+					debug.requests.push(_url);
+				}
+		
+				if (stripResources) {
+					var i = 0;
+					var l = stripResources.length;
+					// /http:\/\/.+?\.(jpg|png|svg|gif)$/gi
+					while (i < l) {
+						if (stripResources[i++].test(_url)) {
+							if (outputDebug) {
+								debug.stripped.push(_url);
+							}
+							request.abort();
+							return;
+						}
+					}
+				}
+		
+				request.continue();
+			});	
+		}
 
-	var options = {};
+		page.on("pageerror", function(err) {  
+			outputError("PAGE ERROR", err); 
+		});
 
-	if (matchMQ) {
-		options.matchMQ = true;
-	}
+		page.on("error", function (err) {  
+			outputError("PAGE ERROR", err);
+		});
+
+		if(!await loadPage() || !await injectCssExtractor()) {
+			await closePuppeteer();
+		}
 
-	if (required) {
-		options.required = required;
+	}
+	catch (e) {
+		outputError("EXCEPTION", e);
+		try {
+			await closePuppeteer();
+		} catch (err) {}
 	}
 
-	if (localStorage) {
-		page.evaluate(function (data) {
-			var storage = window.localStorage;
-			if (storage) {
-				for (var key in data) {
-					storage.setItem(key, data[key]);
-				}
+	return;
+
+	async function loadPage() {
+
+		if (url) {
+	
+			debug.loadTime = new Date();
+	
+			await page.goto(url);
+
+			return true;
+		}
+		else {
+		
+			if (!fakeUrl) {
+				stderr("Missing \"fake-url\" option");
+				return false;
 			}
-		}, localStorage);
-	}
+		
+			html = fs.readFileSync(0, "utf-8");
+		
+			debug.loadTime = new Date();
+	
+			await page.setRequestInterception(true);
+	
+			page.once("request", req => {
+			  req.respond({
+				body: "<!DOCTYPE html><html><head><title>Empty page</title></head><body><div>Empty page</div></body></html>"
+			  });
+			});
+	
+			await page.goto(fakeUrl);
+	
+			if (!stripResources) {
+				// disable request interception to allow caching
+				await page.setRequestInterception(false);
+			}
+	
+			if (diskCacheDir) {
+				await page.setCacheEnabled(true);
+			}
+	
+			await page.setContent(html);
 
-	if (Object.keys(options).length) {
-		page.evaluate(function (options) {
-			window.extractCSSOptions = options;
-		}, options);
+			return true;
+		}	
 	}
 
-	if (!height) {
-		var _height = page.evaluate(function () {
-			return document.body.offsetHeight;
-		});
-		page.viewportSize = {
-			width: width,
-			height: _height
-		};
-	}
+	async function injectCssExtractor() {
 
-	if (!fs.isFile(scriptPath)) {
-		fail("Unable to locate script at: " + scriptPath);
-	}
+		if (!html) {
+			html = await page.content();
```

Comment:
This returns the HTML for the _evaluated_ DOM - which means any injected elements (ad/tracking/statistics scripts) are included - and thus will get _baked_ in (if you're not using the `--css-only` option). So you would probably end up loading ad/tracking/statistics twice :)

That's why the old script loaded the raw HTML via XHR :)
https://github.com/drdk/dr-css-inliner/blob/master/index.js#L287-L296

Reply:
Good point. The old code gave me some trouble, but I will try to make it work.